web-registry/src/lib/ecocredit/api.ts

```diff
@@ -247,22 +300,47 @@ export const getBatchesByProjectWithSupply = async (
     client,
     request: { projectId },
   });
-  const batchesWithData = await addDataToBatch({ batches: batches?.batches });
+  const batchesWithData = await addDataToBatches({ batches: batches?.batches });
   return { data: batchesWithData };
 };
 
-/** Adds Tx Hash and supply info to batch for use in tables */
-type AddDataToBatchParams = {
-  batches: BatchInfo[];
-  sanityCreditClassData?: AllCreditClassQuery;
-  withAllData?: boolean;
+const getClassProjectForBatch = async (
+  batch: BatchInfo,
+  sanityCreditClassData?: AllCreditClassQuery,
+): Promise<ClassProjectInfo> => {
+  const { projectId } = batch;
+  const { project } = await getProject(projectId);
+  let metadata;
+  if (project?.metadata.length) {
+    try {
+      metadata = await getMetadata(project.metadata);
+    } catch (error) {}
+  }
+
+  // TODO should we use credit class metadata instead of Sanity eventually?
```

Comment:
i think so, i've been wondering about this since i've been working in regen web more. i've seen that we still have that dependency in at least a few places.

Reply:
will add a ref to the related task https://github.com/regen-network/regen-registry/issues/1428